main.tf

```diff
@@ -84,3 +84,40 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
   target_resource_instance_id = var.existing_kms_instance_guid
   roles                       = ["Reader"]
 }
+
+##############################################################################
+# Context Based Restrictions
+##############################################################################
+module "cbr_rule" {
+  count            = length(var.cbr_rules) > 0 ? length(var.cbr_rules) : 0
+  source           = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-rule-module?ref=v1.2.0"
+  rule_description = var.cbr_rules[count.index].description
+  enforcement_mode = var.cbr_rules[count.index].enforcement_mode
+  rule_contexts    = var.cbr_rules[count.index].rule_contexts
+  resources = [{
+    attributes = [
+      {
+        name     = "accountId"
+        value    = var.cbr_rules[count.index].account_id
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceInstance"
+        value    = ibm_resource_instance.es_instance.guid
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceName"
+        value    = "messagehub"
+        operator = "stringEquals"
+      }
+    ]
+  }]
+  operations = [{
+    api_types = [
+      {
+        api_type_id = "crn:v1:bluemix:public:context-based-restrictions::::api-type:data-plane"
```

Comment:
@Khuzaima05 On the error that you mentioned at https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/28#issuecomment-1564208222 is working as expected.  There are no supported API Types for messagehub service for CBR integration.  You are trying to add an API type when its not supported.  Please refer to the example https://github.com/terraform-ibm-modules/terraform-ibm-cbr/blob/9935daa1d5d70ca7a51601e54a4bfbe2fa1c023c/examples/multizone-rule/main.tf#L146 to provide an empty operations.

Reply:
resolved